### PR TITLE
ADAL to MSAL - Multiple resources C#

### PR DESCRIPTION
**Pull request by cristobal-buenrostro**
Code sample updates:

- use MSAL instead of ADAL for getting AAD token.
- upgrade to .Net Core 6.

### CODE REVIEW

File: js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs [hunk 1]
```diff
@@ -1,142 +1,144 @@
...
-		/// Get an Azure AD authentication token using a given resource
-		/// </summary>
-		private async Task<string> GetTokenAsync(ImmersiveReaderResourceConfig resourceConfig)
+        /// <summary>
+        /// Get an Azure AD authentication token using a given resource
+        /// </summary>
+        private async Task<string> GetTokenAsync(ImmersiveReaderResourceConfig resourceConfig)
         {
             string authority = $"https://login.windows.net/{resourceConfig.TenantId}";
             const string resource = "https://cognitiveservices.azure.com/";
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs:111]: you renamed to resourceId in other PR

File: js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs [hunk 2]
```diff
@@ -1,142 +1,144 @@
...
-
-
-		public IActionResult Index()
+    {
+        // These should be the keys in the secrets.json file, explained in ReadMe.txt
+        private readonly HashSet<string> ResourceKeys = new HashSet<string>() { "ImmersiveReaderResourceOne", "ImmersiveReaderResourceTwo" };
+        // Map the resource key to the configuration object
+        private readonly Dictionary<string, ImmersiveReaderResourceConfig> ResourceKeyToConfigs;
+
+        IConfidentialClientApplication app;
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs:16]: make this a property

File: js/samples/advanced-csharp-multiple-resources/Models/ErrorViewModel.cs [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
-using System;
-
-namespace QuickstartSampleWebApp.Models
+namespace MultipleResourcesSampleWebApp.Models
 {
     public class ErrorViewModel
     {
-        public string RequestId { get; set; }
+        public string? RequestId { get; set; }
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/Models/ErrorViewModel.cs:8]: revert

File: js/samples/advanced-csharp-multiple-resources/MultipleResourcesSampleWebApp.csproj [hunk 4]
```diff
@@ -1,16 +1,13 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <UserSecretsId>138ab20d-9022-4634-ae3b-4697bc9d5d31</UserSecretsId>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/MultipleResourcesSampleWebApp.csproj:5]: remove nullable?

File: js/samples/advanced-csharp-multiple-resources/Views/Home/Index.cshtml [hunk 5]
```diff
@@ -6,80 +6,86 @@
...
+                cognitiveAppId: 'MultipleResourcesSampleWebApp',
+                onExit: exitCallback
+            };
 
-			getImmersiveReaderLaunchConfigAsync(resourceId)
-				.then((launchParams) => {
+            getImmersiveReaderLaunchConfigAsync(resourceId)
+                .then((launchParams) => {
 
 					const token = launchParams.token;
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/Views/Home/Index.cshtml:62]: fix indent on these

File: js/samples/advanced-csharp-multiple-resources/Views/Shared/Error.cshtml [hunk 6]
```diff
@@ -6,10 +6,10 @@
 <h1 class="text-danger">Error.</h1>
 <h2 class="text-danger">An error occurred while processing your request.</h2>
 
-@if (Model.ShowRequestId)
+@if (Model?.ShowRequestId ?? false)
 {
     <p>
-        <strong>Request ID:</strong> <code>@Model.RequestId</code>
+        <strong>Request ID:</strong> <code>@Model?.RequestId</code>
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/Views/Shared/Error.cshtml:12]: is the Model?. needed?

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs (hunk 1 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs:111]: yes, I will keep as resource in all samples

File: js/samples/advanced-csharp-multiple-resources/Models/ErrorViewModel.cs (hunk 3 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp-multiple-resources/Models/ErrorViewModel.cs]: done

File: js/samples/advanced-csharp-multiple-resources/MultipleResourcesSampleWebApp.csproj (hunk 4 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp-multiple-resources/MultipleResourcesSampleWebApp.csproj]: removed

File: js/samples/advanced-csharp-multiple-resources/Views/Shared/Error.cshtml (hunk 6 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp-multiple-resources/Views/Shared/Error.cshtml]: done

File: js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs [hunk 7]
```diff
@@ -1,142 +1,143 @@
...
+        /// Get an Azure AD authentication token using a given resource
+        /// </summary>
+        private async Task<string> GetTokenAsync(ImmersiveReaderResourceConfig resourceConfig)
         {
             string authority = $"https://login.windows.net/{resourceConfig.TenantId}";
             const string resource = "https://cognitiveservices.azure.com/";
 
-            AuthenticationContext authContext = new AuthenticationContext(authority);
-            ClientCredential clientCredential = new ClientCredential(resourceConfig.ClientId, resourceConfig.ClientSecret);
+            if (app == null)
```
Comment (rwallerms) [inline js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs:112]: add this init code to the Property

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs (hunk 7 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs]: @rwallerms 
init code added to the property

I had to move all tenant,secret, subdomain and clientId vars to class level in order to have them available for the property.

**Review by rwallerms — APPROVED** [review]

**Pull request merged**